lib/classes/ssl/class.lescript.php

```diff
@@ -517,6 +528,13 @@ public function getLastLinks()
 		preg_match_all('~Link: <(.+)>;rel="up"~', $this->lastHeader, $matches);
 		return $matches[1];
 	}
+
+	public function getAgreementURL()
+	{
+		preg_match_all('~Link: <(.+)>;rel="terms-of-service"~', $this->lastHeader, $matches);
+		return $matches[1][0];
```

Comment:
> If the server wishes to present the client with terms under which the ACME service is to be used, it MUST indicate the URI where such terms can be accessed in a Link header with link relation “terms-of-service”. As noted above, the client may indicate its agreement with these terms by updating its registration to include the “agreement” field, with the terms URI as its value.

IMO this means, the URI is optional (in theory, unlikely to encountet this with letsencrypt.org though).
Therefore the matches might not be populated here.

If the URI is not contained, it should also not be set in `postRegAgreement()` above.

I do not know letsencrypt indepth, this is just a concern that arose from reading the specs.